app/controllers/admin/badges_controller.rb

```diff
@@ -19,10 +19,18 @@ def create
 
       if @badge.save
         flash[:success] = "Badge has been created!"
-        redirect_to admin_badges_path
+        if FeatureFlag.enabled?(:admin_restructure)
```

Comment:
When I test this locally, if I remove the `if` statement and only use the old route no matter what, it will properly load the page, and everything is correct except the selection in the left sidebar navigation, which is not "zoomed in" on the Badges drop down from the content manager. 

Is it possible that we are fine to use the old routes but merely need the javascript(I think that is what is triggering that sidebar selection) to recognize the old routes and make the proper selection when the feature flag is turned on? 

<img width="1313" alt="Screen Shot 2021-03-23 at 4 43 37 PM" src="https://user-images.githubusercontent.com/1813380/112215619-91d40280-8bee-11eb-8159-126e5449f4ca.png">

FWIW I could totally be missing something so please feel free to tell me to be quiet if that is the case!

Reply:
@mstruve however, if you look at the URL in your second screenshot, it's pointing to `/admin/badges` instead of `/admin/content_manager/badges` (like in your first screenshot). Ideally we want all our routes, and nested routes (singular or otherwise) to always be scoped under the content_manager. This will ensure that users do not get lost in the application.

Reply:
@mstruve The other thing is - in a weeks time, when we remove the old routes in favour of the new ones, it would definitely make it easier to remove if we have the feature flags scattered in the codebase already. That way we'd be removing the `else` and just keeping the contents of the `ifs`. I also feel like the more laborious/manual way of adding the feature flags everywhere allows us to more accurately test the end result, without any mechanism in between. 

With that being said, there is that tradeoff with the clutter of scattering the feature flags everywhere.

Reply:
@Ridhwana I'm speaking for Molly at her request since she's in the middle of recording a podcast :-)

>  if you look at the URL in your second screenshot, it's pointing to `/admin/badges` instead of `/admin/content_manager/badges`

I see the RFC prescribes nesting the URL path under `content_manager` and I think Molly was asking if that's necessary. Something along the lines of this:

```ruby
# config/routes.rb

constraints(proc { FeatureFlag.enabled? :admin_restructure }) do
  # route block pointing to the new controllers without the `scope :path` call
end

# route block pointing to the old controllers
```

The links wouldn't need to change and removing the feature flag would (theoretically! :joy:) be a simple matter of removing the old block and unwrapping the new block out of its constraints and not worrying about all of the links all over the application.

_However_, I can see how keeping the `scope :path` call could be useful, in the sense of nesting routes going hand-in-hand with nesting the UI layers. IIRC, this was a big tenet of Ember.js routing and has become common in a lot of other front-end routing libraries. I feel like the disconnect is that there's no concept of those intermediate routes in the Rails router. For example, `/admin/content_manager` doesn't seem to route to anything.

Reply:
To be clear, I may very well be way off-base here with my reading of the discussion. I was admittedly _also_ curious about adding the new routes vs routing the same URLs to different controllers, but was trying to stay neutral because we're making changes and there's a lot about this that I haven't even seen how it _used to_ work yet and I've learned more about this admin restructure in the past hour than I have in the past 3 weeks. 😄

Reply:
@jgaskins I appreciate you jumping in and clarifying 😊

 > I see the RFC prescribes nesting the URL path under content_manager and I think Molly was asking if that's necessary.

If you look at the `master` branch at the moment and toggle the feature flag to true, it works with the top level routes nested under the scopes like 'content_manager` and `customization` etc, however the nested routes continue working as per the old routes (w/o scopes). This is I assume teh way that Molly is suggesting it could work. 

Personally, the answer to this question at 11:55pm (this may change after reading the thread in the morning 😅 ) is that is is necessary in order for us to have a smooth user experience + to allow us to know where we are in the routing structure and to set the correct current nav items (which I've not implemented as yet).
Going back and forth between pages and clicking on filters feels strange on master at the moment for this reason.

> I feel like the disconnect is that there's no concept of those intermediate routes in the Rails router. For example, /admin/content_manager doesn't seem to route to anything.
However, I do agree that this does leave me with an unsettling feeling, but there's no concept of those intermediate routes in the Rails router because the routes are corresponding to the UI whereby we don't map anything to that route. 

The part that is confusing to me in the discussion is the codeblock is the comment:

>  # route block pointing to the new controllers without the `scope :path` call and 
> #  route block pointing to the old controllers

plus this: 
> I was admittedly also curious about adding the new routes vs routing the same URLs to different controllers,

There is no concept of the old controllers and new controllers because we're simply reusing the existing controllers from the old routes for the scoped routes. The  only thing we've added is the scope in the feature flagged routes to play nicely with the way the UI is nested in the RFC.

To take a step back I think the clarity I need at the moment to move forward is how do we want the nested routes to look as there seems to be uncertainty about this:

Would it work like this (option 1)?
- Customization> HTML Variants can be found at http://localhost:3000/admin/customization/html_variants 
- The singular HTML variant route when you click on an item on the table should route to  http://localhost:3000/admin/customization/html_variants/1

or like this(option 2)?
- Customization> HTML Variants can be found at http://localhost:3000/admin/customization/html_variants 
- The singular HTML variant route when you click on an item on the table should route to  http://localhost:3000/admin/html_variants/1


I guess if we do proceed with this option 1 then I need to figure out how do we reduce the scatter of the Feature Flags across the codebase by moving that logic into the routes possibly. However, I feel like redirecting the old routes to the new routes will add technical debt in the codebase and we should just remove traces of the old routes and use the new scoped routes directly. 


I hope I didn't write complete garbage above or completely misunderstand your points.

Reply:
@Ridhwana Thank you for all of this, this clears up a lot!

> There is no concept of the old controllers and new controllers because we're simply reusing the existing controllers from the old routes for the scoped routes. The reason that I've added the scope in the feature flagged routes is so that we didn't need to rewrite the controller system.

Oh I very much overlooked this part — my apologies. If this is a matter of pointing new URLs the same code, this makes _a lot_ of sense. Without some unnervingly clever routing hacks to get the same routing helpers pointing at the new paths, we _absolutely_ have to change the links everywhere in that case to ensure that they point at the new URLs. 🙌🏼 /cc @mstruve

Reply:
@jgaskinsno need to apologise, thanks for taking the time yesterday to clarify and talk through the issue! Due to yours, molly's and Dan's comments, I discovered that I was adding `as` to my scoped route which was changing the path helpers - finding that bug and removing it made a huge difference.  

I've been working on a part 2 https://github.com/forem/forem/pull/13114. Whilst I'm still hitting some blockers with path helpers (which I'll provide more info about in that PR), it seems more promising :)